File: Components/Dashboard/CS/DashboardMainDemo/Dashboards/SalesDetails.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.DataAccess;
using DevExpress.DashboardCommon;

namespace DashboardMainDemo.Dashboards {
    public partial class SalesDetails : Dashboard {
        public SalesDetails() {
            InitializeComponent();
        }
        void SalesDetails_DataLoading(object sender, DashboardDataLoadingEventArgs e) {
            SalesDetailsDataGenerator dataGenerator = new SalesDetailsDataGenerator(DataLoader.LoadSales());
            dataGenerator.Generate();
            e.Data = dataGenerator.Data;
        }
    }
}

File: Components/Dashboard/VB/DashboardMainDemo/Dashboards/SalesDetails.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.DataAccess
Imports DevExpress.DashboardCommon

Namespace DashboardMainDemo.Dashboards
	Partial Public Class SalesDetails
		Inherits Dashboard
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub SalesDetails_DataLoading(ByVal sender As Object, ByVal e As DashboardDataLoadingEventArgs) Handles MyBase.DataLoading
			Dim dataGenerator As New SalesDetailsDataGenerator(DataLoader.LoadSales())
			dataGenerator.Generate()
			e.Data = dataGenerator.Data
		End Sub
	End Class
End Namespace